src/snowcli/cli/stage/commands.py

```diff
@@ -66,12 +65,11 @@ def stage_get(
 def stage_put(
     path: Path = typer.Argument(
         ...,
-        exists=True,
-        file_okay=True,
-        dir_okay=True,
-        writable=True,
-        resolve_path=True,
-        help="File or directory to upload to stage",
+        exists=False,
```

Comment:
Hmmm, at the beginning I was wondering if it's a good solution. I thought that maybe we should accept a list of files and let a shell do glob expansion (snow `stage put ./*.py stageName` translated to list of paths in Typer arg). But then I looked at [SQL docs for PUT operation](https://docs.snowflake.com/en/sql-reference/sql/put) and I couldn't find any way to upload these multiple files without a glob pattern, what can mean that we cannot expand a pattern before passing it to SQL.

So, if I understand it correctly (please correct me if you see sth wrong), then:
* Your current solution will work for `snow stage put "./*.py" stageName` (path in quotes) on MacOS and Linux
* Your current solution may not work for `snow stage put "./*.py" stageName` (path in quotes) on Windows
   * I didn't test it but I found GitHub discussion and PR to Click (base framework for Typer) adding custom glob extension for Windows:
     * https://github.com/pallets/click/issues/1096
     * https://github.com/pallets/click/pull/1830
* Your current solution may not work for `snow stage put ./*.py stageName` (path not in quotes)
   * I didn't test it but I think it won't work because:
      * Linux and MacOS shells will do glob expansion before passing arguments to SnowCLI
      * Click may do some magic on Windows
* Previous implementation couldn't allow for glob patterns but worked for both quoted and unquoted paths.

IMO we should confirm how your implementation works for the following cases:
* Linux / MacOS + path with glob pattern and without quotes
* Windows + path with glob pattern and with quotes
* Windows  + path with glob pattern and without quotes

Reply:
I forgot to write that at the moment we need to test these cases manually because our tests use Click's CliRunner, so they don't execute real commands from real shells. It's good enough for most cases but in this case IMO we have to be sure how it really works in real shell.

Reply:
I can confirm that it does not work on mac without quotes due to the expansion you mentioned. I don't have a good way to test it on Windows, but agree that we should test if the behavior works on Windows. My guess is that it would work on Windows with or without quotes, since Windows doesn't expand glob-style paths by default, but I don't know for sure how Click handles paths like that.

My take on this is that, it would be great to support all of the cases, but at a minimum this doesn't remove any old functionality, and at last on Mac OS allows for new functionality (which I need for my use-case).

One reason why this is particularly important is because of the unexpected (to me) behavior of doing stage put with a folder name -- if the folder has other folders nested in it, it fails, because it tries to upload all of the nested folders as well as the files.

Reply:
I understand your point of view, but I am afraid of adding a functionality which may not work on all platforms and which may be not so easy to use (and to document) because of possible different behaviours described above.

The first issue may be resolved by tests (probably manual tests at the moment). 

The second one is more difficult. Most CLI tools usually do not need quotes to use glob patterns for files. Even more - users cannot use quotes because shell won't expand the pattern if they do. 
Maybe a good help message could be a solution. Such which will mention that you to have use quotes when using a glob pattern for this argument. And such which will include info about Windows case if it turns out that glob patterns don't work on Windows.

@sfc-gh-turbaszek What do you think about it?

Reply:
One more note about Windows:

Generally you shouldn't have to figure out how to test it on windows. It should be on us - because it won't be the last PR with changes which may not work on Windows and we have to take care of it. And we have planned a task to add support for Windows in automatic tests executed from GitHub actions but unfortunately we haven't taken care of it yet.

Reply:
@sfc-gh-zblackwood 

Ok, I talked with @sfc-gh-turbaszek and we decided to follow your take on this and merge the PR as it is, even without tests of glob patterns on Windows. Let's just change the help text for the argument to make it clear that users have to enclose the path in quotes when using glob patterns.

Reply:
Updated the note on the field -- how does that look?